extensions-core/druid-basic-security/src/main/java/io/druid/security/basic/db/BasicSecurityStorageConnector.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.security.basic.db;
+
+import java.util.List;
+import java.util.Map;
+
+public interface BasicSecurityStorageConnector
```

Comment:
What do you think about making all non get* methods to return the result indicating success or failure? For example, delete* methods return the number of deleted rows. It may be useful for operators to reduce mistakes.

Reply:
I ended up using exceptions to communicate invalid accesses to the caller, felt like that was clearer than returning an "affected rows count" (e.g., suppose I delete a role from a user and get 0 as a return value, it could represent multiple conditions like user not found, role not found, or user doesn't have role, the exceptions I throw have exact messages for the failure cause)